### PR TITLE
Adjustable PoW difficulty

### LINKED ISSUE #138: Contact authority for PoW difficulty
We want to be able to have a DNS entry that snodes can ask for the current PoW difficulty
This would allow us to do things like hitting a "panic switch" to crank up the PoW difficulty if we detect that the network is being flooded

The storage server should poll (every 10 mins?) this authority and update a local difficulty value it uses to validate message PoW

If a message is received that has invalid PoW, the snode should respond with a 4XX specific for bad PoW difficulty and the correct difficulty value

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Every 10 mins the snodes will query the sentinel dns text record to get their PoW difficulty (previously NONCE_TRIALS)
Snodes will now send the difficulty with every successful request to allow clients to lower their difficulty if needed
Snodes will now send the difficulty with a 402: payment_required response when the clients provide invalid PoW to allow them to raise their difficulty if needed

Also added a lil bit to vscode settings to make my vscode experience a lil nicer and I think there was a couple straggler clang format changes that weren't caused by my changes

closes #138

### CODE REVIEW

File: httpserver/service_node.cpp [hunk 1]
```diff
@@ -13,11 +13,13 @@
 #include <chrono>
 #include <fstream>
 #include <iomanip>
+#include <resolv.h>
```
Comment (msgmaxim) [inline httpserver/service_node.cpp]: This is not portable, is it?

File: httpserver/http_connection.cpp [hunk 2]
```diff
@@ -614,13 +614,19 @@ void connection_t::process_store(const json& params) {
...
     const bool validPoW =
-        checkPoW(nonce, timestamp, ttl, pubKey, data, messageHash);
+        checkPoW(nonce, timestamp, ttl, pubKey, data, messageHash,
+                 service_node_.get_pow_difficulty());
 #ifndef DISABLE_POW
     if (!validPoW) {
-        response_.result(http::status::forbidden);
-        response_.set(http::field::content_type, "text/plain");
-        body_stream_ << "Provided PoW nonce is not valid.\n";
+        response_.result(http::status::payment_required);
```
[the remark this answers is not in the document]
Comment (BeaudanBrown) [reply httpserver/http_connection.cpp]: Replaced with boring number :^(

File: httpserver/service_node.cpp (hunk 1 above)
Comment (BeaudanBrown) [reply httpserver/service_node.cpp]: After talking to Simon, plan is to leave windows support on TODO list

File: httpserver/service_node.cpp [hunk 3]
```diff
@@ -367,12 +397,23 @@ void ServiceNode::on_swarm_update(const block_update_t& bu) {
     initiate_peer_test();
 }
 
+void ServiceNode::pow_difficulty_timer_tick() {
+    int new_difficulty = query_pow_difficulty();
```
Comment (sachaaaaa) [inline httpserver/service_node.cpp]:
```suggestion
    const int new_difficulty = query_pow_difficulty();
```

File: httpserver/service_node.cpp [hunk 4]
```diff
@@ -105,7 +133,7 @@ static std::shared_ptr<request_t> make_push_request(std::string&& data) {
     return make_post_request("/v1/swarms/push", std::move(data));
 }
 
-static bool verify_message(const message_t& msg,
+static bool verify_message(const message_t& msg, const int pow_difficulty,
```
Comment (sachaaaaa) [inline httpserver/service_node.cpp]:
```suggestion
static bool verify_message(const message_t& msg, int pow_difficulty,
```

File: httpserver/service_node.cpp [hunk 5]
```diff
@@ -29,6 +31,31 @@ constexpr std::array<std::chrono::seconds, 6> RETRY_INTERVALS = {
...
+int query_pow_difficulty() {
+    int response;
+    unsigned char query_buffer[1024] = {};
+    response = res_query(POW_DIFFICULTY_URL, ns_c_in, ns_t_txt, query_buffer,
+                         sizeof(query_buffer));
+    int pow_difficulty;
+    ns_msg nsMsg;
+    ns_initparse(query_buffer, response, &nsMsg);
+    ns_rr rr;
+    ns_parserr(&nsMsg, ns_s_an, 0, &rr);
```
Comment (sachaaaaa) [inline httpserver/service_node.cpp]: Just wondering if this can ever throw?

Comment (msgmaxim) [reply httpserver/service_node.cpp]: > ns_parserr returns zero on success and -1 if it fails to parse the response buffer

Comment (BeaudanBrown) [reply httpserver/service_node.cpp]: Added error handling

**Pull request merged**